source/h1_encoder.c

```diff
@@ -551,12 +551,13 @@ struct aws_h1_chunk *aws_h1_chunk_new(struct aws_allocator *allocator, const str
     chunk->user_data = options->user_data;
     chunk->chunk_line = aws_byte_buf_from_empty_array(chunk_line_storage, chunk_line_size);
     s_populate_chunk_line_buffer(&chunk->chunk_line, options);
-
+    aws_input_stream_acquire(chunk->data);
```

Comment:
trivial: do it like `chunk->data = aws_input_stream_acquire(options->chunk_data)` (needs to happen a few lines up)